qiita_db/sql_connection.py

```diff
@@ -402,9 +430,16 @@ def execute_fetchall(self, sql, sql_args=None):
         """
         with self._sql_executor(sql, sql_args) as pgcursor:
             result = pgcursor.fetchall()
-        return result
+            if return_types:
```

Comment:
Second option is good and doesn't introduce ambiguity in return types
On Mar 10, 2015 3:23 PM, "Joshua Shorenstein" <notifications@github.com>
wrote:

> In qiita_db/sql_connection.py
> <https://github.com/biocore/qiita/pull/930#discussion_r26172710>:
>
> > @@ -402,9 +430,16 @@ def execute_fetchall(self, sql, sql_args=None):
> >          """
> >          with self._sql_executor(sql, sql_args) as pgcursor:
> >              result = pgcursor.fetchall()
> > -        return result
> > +            if return_types:
>
> First option would be painful and require changing every execute_fetch*
> call. Second option is doable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/930/files#r26172710>.
>